src/ansys/sound/core/psychoacoustics/spectral_centroid.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (C) 2023 - 2025 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Computes the spectral centroid of the signal."""
+import warnings
+
+from ansys.dpf.core import Field, Operator
+import numpy as np
+from numpy import typing as npt
+
+from . import PsychoacousticsParent
+from .._pyansys_sound import PyAnsysSoundException, PyAnsysSoundWarning
+
+
+class SpectralCentroid(PsychoacousticsParent):
+    """Computes the spectral centroid of a signal."""
+
+    def __init__(self, signal: Field = None):
+        """Create a ``SpectralCentroid`` object.
+
+        Parameters
+        ----------
+        signal: Field
+            Signal in Pa to compute spectral centroid on as a DPF field.
```

Comment:
@anshlachamb I could imagine some relevance in computing SC for non-Pa signals too (because it is a not psychoacoustics indicator per se). I suggest to remove "in Pa" here.

Reply:
done